Чтение и показ записей из MySQL

delannushka opened the pull request:
---
:mortar_board: [Чтение и показ записей из MySQL](https://up.htmlacademy.ru/php/14/user/2069609/tasks/11)

delannushka committed:
Выполнила 5.6. Чтение и показ записей из MySQL

nkolosov commented on index.php:
```diff
@@ -1,70 +1,67 @@
+    $content = include_template('error.php', ['error' => $error]);
+}
+else {
+    $sql1 = 'SELECT * FROM projects WHERE user_id = 3';
```
id пользователя должен быть отдельным параметром задаваться из переменной

Кроме того, не стоит использовать переменные с числовыми индексами. 

Если переменная больше не используется, новые данные того же типа можно заново записать в ту же переменную

nkolosov commented on index.php:
```diff
@@ -1,70 +1,67 @@
+        $content = include_template('error.php', ['error' => $error]);
+    }
+}
+
```
Куча лишних пустых строк

delannushka committed:
Использовала подготовленные выражения.

nkolosov commented on index.php:
```diff
@@ -1,69 +1,59 @@
+        }
+        else {
+            $error = mysqli_error($link);
+            $content = include_template('error.php', ['error' => $error]);
```
В случае ошибки, переменная `$content` не используется, ниже всё равно получается, что в шаблон грузится `$page_content` текущей страницы, а ошибка игнорируется

nkolosov replied:
Аналогичная ситуация с другими ошибками на этой странице

nkolosov commented on index.php:
```diff
@@ -1,69 +1,59 @@
+}
+else {
+    $sql = 'SELECT * FROM projects WHERE user_id = ?';
+    $stmt = mysqli_prepare($link, $sql);
```
Эту конструкцию (и ниже) можно заменить на вызов функции, которая уже есть в `helpers.php`

```
$stmt = db_get_prepare_stmt($link, $sql, [$current_user_id]);
$result = mysqli_stmt_get_result($stmt);
....
```

delannushka committed:
Исправила вывод контента на страницу.
Менять на готовую функцию db_get_prepare_stmt не стала, тк выдает ошибку: Fatal error: Uncaught mysqli_sql_exception: Commands out of sync; you can't run this command now in C:\OpenServer\domains\doingsdone\index.php:33 Stack trace: #0 C:\OpenServer\domains\doingsdone\index.php(33): mysqli_stmt_get_result() #1 {main} thrown in C:\OpenServer\domains\doingsdone\index.php on line 33
